vllm/v1/offloading/cpu.py

```diff
@@ -54,3 +66,64 @@ def allocate_blocks(self, block_hashes: list[int]) -> list[BlockStatus]:
     def free(self, block: BlockStatus):
         assert isinstance(block, CPUBlockStatus)
         self.allocated_blocks_free_list.append(block.block_id)
+
+
+class CPUOffloadingSpec(OffloadingSpec):
+
+    def __init__(self, vllm_config: VllmConfig):
+        super().__init__(vllm_config)
+
+        self._num_cpu_blocks: Optional[int] = None
+
+        # scheduler-side
+        self._manager: Optional[OffloadingManager] = None
+
+        # worker-side
+        self._handler: Optional[OffloadingHandler] = None
+
+    @property
+    def num_cpu_blocks(self):
+        if self._num_cpu_blocks is None:
+            self._num_cpu_blocks = self.extra_config.get("num_cpu_blocks")
+            if not self._num_cpu_blocks:
+                raise Exception("num_cpu_blocks must be specified "
+                                "in kv_connector_extra_config")
+        return self._num_cpu_blocks
+
+    def get_manager(self) -> OffloadingManager:
+        if not self._manager:
+            kv_events_config = self.vllm_config.kv_events_config
+            enable_events = (kv_events_config is not None
+                             and kv_events_config.enable_kv_cache_events)
+            self._manager = LRUOffloadingManager(CPUBackend(
+                block_size=self.offloaded_block_size,
+                num_blocks=self.num_cpu_blocks),
+                                                 enable_events=enable_events)
+        return self._manager
+
+    def get_handlers(
+        self, kv_caches: dict[str, torch.Tensor]
+    ) -> Iterator[tuple[type[LoadStoreSpec], type[LoadStoreSpec],
+                        OffloadingHandler]]:
+        if not self._handler:
+            if not current_platform.is_cuda():
+                raise Exception("CPU Offloading is currently only supported"
+                                " on CUDA GPUs")
+            attn_backend = get_attn_backend(
+                self.vllm_config.model_config.get_head_size(),
+                self.vllm_config.model_config.dtype,
+                self.vllm_config.cache_config.cache_dtype,
+                self.gpu_block_size,
+                self.vllm_config.model_config.is_attention_free,
+                use_mla=self.vllm_config.model_config.use_mla)
```

Comment:
I feel like we should add a `get_attn_backend_from_config(VllmConfig)` and use that both here and in the `NixlConnector`.